### PR TITLE
SECD-746 Custom stats not emitting due to async mode

### PR DESCRIPTION
**Pull request by bkrebsbach**
In the async case of [serverfull's invoke api](https://github.com/asecurityteam/serverfull/blob/master/invokeapi.go#L117), the base context is wrapped,
but when it gets cancelled, all parent/children contexts are unlinked. This
causes the logger/stater that's injected to fail. Disabling async mode for now.

Also adding a check for the status code of the first call to Nexpose.

### CODE REVIEW

File: pkg/assetfetcher/http.go [hunk 1]
```diff
@@ -82,6 +82,12 @@ func (c *NexposeAssetFetcher) FetchAssets(ctx context.Context, siteID string) (<
 		return nil, errChan
 	}
 
+	if res.StatusCode != http.StatusOK {
```
Comment (kconwayatlassian) [inline pkg/assetfetcher/http.go]: question: Did you mean for this content to be included? It does not appear related to the stats problem

Comment (bkrebsbach) [reply pkg/assetfetcher/http.go]: True, I figured it was small enough to add into this PR. I'll separate it out into it's own PR.

Comment (bkrebsbach) [reply pkg/assetfetcher/http.go]: Fixed.
